internal/tuf/tuf_test.go

```diff
@@ -245,3 +252,67 @@ func TestDelegationMatches(t *testing.T) {
 		assert.Equal(t, test.expected, got, fmt.Sprintf("unexpected result in test '%s'", name))
 	}
 }
+
+func TestRootMetadataWithSSHKey(t *testing.T) {
```

Comment:
This test function is a bit long and maybe too ssh-specific for `tuf_test.go`. I just wanted to include a metadata integration POC / usage example. Happy to drop the commit, if this feels misplaced.

Reply:
I think it's fine tbh. We can revisit when we have other signers / key types we want to test similarly.